### DISCUSSION TITLE
How to change output directory?

### DISCUSSION THREAD

**Discussion by ccgauvin94** [nextpyp/.github#122] [Q&A]
Hi,

I'd love to give NextPYP a shot for live data analysis. From playing around with smaller datasets, it's working quite well. However, and I hope this isn't glaringly obvious and I'm just missing it, I can't seem to find a way to change the output directory of the processed data. Is there any way, either through the GUI or command line, to direct pyp as to where the processed data should wind up, on a per-session basis? 

Thanks

**Comment by abartesaghi** [comment]
All sessions are currently saved under the `sessions/` directory in the `shared/` folder provided in the configuration file (there is button next to each session that will tell you the absolute path for each session). So, no, there is currently no way to specify an output directory for each session. Can you explain the use case you have and/or say if you want to save just the raw data to a specific directory or also the pre-processing results?

**Reply by ccgauvin94** [reply]
Hi Alberto,

Sorry for the delayed response here. The use case is that we are running NextPYP on the cloud using (in our case) Google Filestore. The storage is ephemeral. Once the data are processed, we download the map and archive the entire filestore, organized by program. 

One possibility would be to keep a single large filestore to just dump data into for live processing, and once done, move the folder out from the sessions folder to the archive when done. I'm not 100% sure how NextPYP will handle that if the files are missing, but I'll give it a shot. That said I imagine this will be a a relatively common use case when running on the cloud, considering the cost of cloud storages. 

Ideally, we would move the entire session (raw data and processed data, although usually we delete the raw data after processing unless we have some specific reason to keep it).

**Reply by abartesaghi** [reply]
Removing the raw data after a session ends will not have any noticeable effect from the GUI, just that you won't be able to re-process the data using different settings, for example. It is useful to know about this use-case though and we will take this into consideration going forward.
